### PR TITLE
Fix Python unit tests under Python 3

### PR DESCRIPTION
**Pull request by martyngigg**
Fixes all Python unit tests so that they are compatible with Python 3.

**To test:**

The build servers will check the Python 2 unit tests so they should still work.To check for Python 3,
build the framework against Python 3 and run `ctest -R ^PythonInterface` -  all of these tests should now pass.

No issue number.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by rosswhitfield** [comment]
I have one test sporadically failing  `PythonInterfaceAPI_ITableWorkspaceTest`. You need to run it a few times before it fails.
```
432: /home/rwp/mantid/Framework/PythonInterface/test/python/mantid/api/ITableWorkspaceTest.py:76: DeprecationWarning: Please use assertEqual instead.
432:   self.assertEquals(table.columnCount(), 1)
432: .EFE/home/rwp/mantid/Framework/PythonInterface/test/python/mantid/api/ITableWorkspaceTest.py:27: DeprecationWarning: Please use assertAlmostEqual instead.
432:   self.assertAlmostEquals(self._test_ws.cell('r_gd_prtn_chrg',0), 10.040912628173828, 15)
432: .........
432: ======================================================================
432: ERROR: test_adding_table_data_using_list (__main__.ITableWorkspaceTest)
432: ----------------------------------------------------------------------
432: Traceback (most recent call last):
432:   File "/home/rwp/mantid/Framework/PythonInterface/test/python/mantid/api/ITableWorkspaceTest.py", line 101, in test_adding_table_data_using_list
432:     table.addRow(values)
432: TypeError: Wrong datatype <str> for column <index> (expected <int>)
432: 
432: ======================================================================
432: ERROR: test_adding_table_data_using_tuple (__main__.ITableWorkspaceTest)
432: ----------------------------------------------------------------------
432: Traceback (most recent call last):
432:   File "/home/rwp/mantid/Framework/PythonInterface/test/python/mantid/api/ITableWorkspaceTest.py", line 118, in test_adding_table_data_using_tuple
432:     table.addRow(values)
432: TypeError: Wrong datatype <str> for column <index> (expected <int>)
432: 
432: ======================================================================
432: FAIL: test_adding_table_data_using_numpy (__main__.ITableWorkspaceTest)
432: ----------------------------------------------------------------------
432: Traceback (most recent call last):
432:   File "/home/rwp/mantid/Framework/PythonInterface/test/python/mantid/api/ITableWorkspaceTest.py", line 140, in test_adding_table_data_using_numpy
432:     self.assertEquals(insertedrow, nextrow)
432: AssertionError: {'value': 1, 'index': 10} != {'value': 10, 'index': 1}
432: - {'index': 10, 'value': 1}
432: ?            -
432: 
432: + {'index': 1, 'value': 10}
432: ?                        +
432: 
432: 
432: ----------------------------------------------------------------------
432: Ran 13 tests in 0.005s
432: 
432: FAILED (failures=1, errors=2)
1/1 Test #432: PythonInterfaceAPI_ITableWorkspaceTest ...***Failed    1.43 sec
```

**Comment by martyngigg** [comment]
That's odd. What version of Python 3 is it?

**Comment by rosswhitfield** [comment]
I see this with python 3.4 on Ubuntu 14.04 and python 3.5 on Arch.

**Comment by martyngigg** [comment]
@rosswhitfield I should have those sporadic failures now.

File: Framework/PythonInterface/test/python/mantid/kernel/ArrayPropertyTest.py [hunk 1]
```diff
@@ -7,16 +7,17 @@
                            NullValidator)
 from mantid.api import PythonAlgorithm
 import numpy as np
+import sys
 
 class ArrayPropertyTest(unittest.TestCase):
 
-    def test_default_constructor_raises_an_exception(self):
+    def xtest_default_constructor_raises_an_exception(self):
```
Comment (rosswhitfield) [inline Framework/PythonInterface/test/python/mantid/kernel/ArrayPropertyTest.py]: Do these tests still pass with python 2? Should they be skipped only for python 3?

Comment (martyngigg) [inline Framework/PythonInterface/test/python/mantid/kernel/ArrayPropertyTest.py]: Nice catch. They should be on. I'd disabled them while debugging a segfault and didn't re-enable them. I'll turn them back on.

**Comment by rosswhitfield** [comment]
👍 :shipit:

**Comment by FedeMPouzols** [comment]
The system tests build failure was totally unrelated to this.

**Pull request merged**